master/docs/manual/credentialmanagement.rst

```diff
@@ -0,0 +1,100 @@
+Credential Management
+=====================
+
+Definition
+----------
+
+Buildbot needs to send and/or execute commands requiring credentials.
+
+The goal is to store credentials used by Buildbot commands.
+
+Credential could not be stored in ``buildbot.config`` or ``master.config`` because the configuration is available in a source control management.
```

Comment:
The framewework provides a way to configure the credentials outside of the master.cfg.
As the master.cfg is usually stored in a SCM like git, those credentials would be available to anybody.
For this reason, the credentials are stored in the buildbot database, optionally encrypted.